### PR TITLE
Updates to typings

### PR DESCRIPTION
**Pull request by raitono**
My Typescript compiler complains if I try to use the DDragon endpoints since they are not included in the typings file.

I've added them, but the only way I could get it to accept the DDragonRequest return type was by converting it to a class. If you would prefer that to not be touched, I can have them return any. I wasn't sure which one was best, but thought it would be better to keep it specific rather than falling back on the any type.

I also added the rune bonuses to the MatchV4ParticipantStatsDto. These are not on the official documentation page yet, but they are returned in the data.

### CODE REVIEW

File: typings/dtos.ts [hunk 1]
```diff
@@ -1231,6 +1231,18 @@ export interface MatchV4ParticipantStatsDto {
     totalHeal?: number // int64
     totalMinionsKilled?: number // int32
     timeCCingOthers?: number // int64
+    /**
+     * Offense rune stat bonus
+     */
+    statPerk0: number // int32
```
Comment (cnguy) [inline typings/dtos.ts]: For this, try running `yarn swagger` to automatically generate the types and let me know if they are the same.

Comment (cnguy) [reply typings/dtos.ts]: Never mind, I'm guessing it doesn't (you already caught it):

> I also added the rune bonuses to the MatchV4ParticipantStatsDto. These are not on the official documentation page yet, but they are returned in the data.

Try running it anyways to see if there is any difference though if you don't mind.

**Comment by cnguy** [comment]
Thank you for the pull request! This is nice.

**DDragonRequest**
I don't mind converting it to a class at all if we are able to restrict the types effectively.

**Adding manual types**
I left one comment regarding using `yarn swagger`, do you mind running it just to double-check? I know the new match stuffs is not on the documentation though, and so it'll probably fail, but it's more of a sanity check.

I do not mind adding manual types, so this is fine.

**Review by cnguy — CHANGES_REQUESTED** [review]
Thank you for the pull request! This is a nice fix for now, while we wait for DDragon typings.

**DDragonRequest**
I don't mind converting it to a class at all if we are able to restrict the types effectively.

**Adding manual types**
I left one comment regarding using `yarn swagger`, do you mind running it just to double-check? I know the new match stuffs is not on the documentation though, and so it'll probably fail, but it's more of a sanity check.

I do not mind adding manual types, so this is fine.

On a side-note, have you tried using your GitHub PR branch typings in your own project?

File: typings/dtos.ts (hunk 1 above)
Comment (raitono) [reply typings/dtos.ts]: Oh, I didn't realize this was auto-generated. I have not used many of the tools here before.

I ran `yarn swagger` and the statPerk fields disappear as expected. The V3 interfaces also get removed as well as some more minor changes to other interfaces.

I took a brief look at the riotapi-schema project which it pulls from. Would it be worth it to look into using a manual DTO override?

**Comment by raitono** [comment]
I did use the DTO typing in my project. Honestly I just want them to make the VS Code Intellisense work :smile:

I have only used the Version endpoint of DDragon, so have nothing written up to test the other bits. My only real concern is that even though I managed to get the properties/fields of the endpoints, the return type is always `any` since it is coming from a plain .js file. I didn't notice this at first. Not sure I can call it a "typing" if they don't return any types!

File: typings/dtos.ts (hunk 1 above)
Comment (cnguy) [reply typings/dtos.ts]: Sorry, I missed this. I'm fine with manual DTO overrides! I want TypeScript support to be better.

**Review by cnguy — APPROVED** [review]

**Comment by cnguy** [comment]
@raitono Is it possible for you to make the branch merge destination `dev` instead of master? If you don't mind :). I will test it tomorrow morning ASAP

**Comment by raitono** [comment]
I now have an open issue with the Riot Developer Relations to get the rune statPerks added to the documentation. Once they do that, swagger should be able to update and get them in officially. If it takes them forever, I may do a pull request to the riotapi-swagger project to get that override in.

**Pull request merged**